src/FSharp.Core/prim-types.fs

```diff
@@ -287,6 +287,7 @@ namespace Microsoft.FSharp.Core
         let mutable maintainsVarSpace = false
         let mutable maintainsVarSpaceWithBind = false
         let mutable joinOnWord = ""
+        new() = CustomOperationAttribute("")
```

Comment:
Code using this new constructor will not work in older versions of the compiler.
Based on the pattern match in customOperationMethods, I guess it will look like it is not a custom operation.

Maybe we should mark this new API ( = the new constructor) as experimental to indicate that to library authors?
At the same time, we should not use in FSharp.Core ourselves, but that is on us to watch for.

Reply:
Been thinking about the same. I will see if it will be too intrusive if used with proper compiler.